agent/consul/state/txn.go

```diff
@@ -383,15 +382,21 @@ func (s *Store) txnDispatch(tx *memdb.Txn, idx uint64, ops structs.TxnOps) (stru
 // is done in a full write transaction on the state store, so reads and writes
 // are possible
 func (s *Store) TxnRW(idx uint64, ops structs.TxnOps) (structs.TxnResults, structs.TxnErrors) {
-	tx := s.db.Txn(true)
+	tx := s.db.WriteTxn(idx)
 	defer tx.Abort()
 
 	results, errors := s.txnDispatch(tx, idx, ops)
 	if len(errors) > 0 {
 		return nil, errors
 	}
 
-	tx.Commit()
+	err := tx.Commit()
+	if err != nil {
+		return nil, structs.TxnErrors{
+			// TODO: What is the right OpIndex? len(ops) or 0?
+			{What: err.Error(), OpIndex: 0},
```

Comment:
I'm not sure what `OpIndex` should be for this error.

Reply:
I think zero makes the most sense. _If_ any clients are actually using this I guess that would provide the most useful experience for end users short of changing the API docs to specify that it can be omitted or a special value means "global error" or something. Both are breaking changes so I'd go for zero in this case - we'd expect errors during event publish to be bugs anyway.